llmfoundry/models/layers/ffn.py

```diff
@@ -139,24 +140,27 @@ def __init__(
             expansion_ratio,
             ffn_hidden_size,
         )
-        self.fc_kwargs: dict[str, Any] = {
-            'bias': bias,
-        }
 
-        self.fc_kwargs['device'] = device
+        if fc_type is None:
+            self.fc_type = copy.deepcopy(fc_type_defaults)
+            self.fc_type['bias'] = bias
+            self.fc_type['device'] = device
+        else:
+            self.fc_type = fc_type
```

Comment:
don't we still need to set bias and device in this case?

Reply:
hmm...I was assuming that MPTBlock init would pass the `fc_type` dict to both ffn and attention, and I modified MPTBlock init to make sure to add in `bias` and `device` keys to the dict. However, that also means that instantiating the FFN or Attention alone (that is, not by way of MPTBlock init) would not add these keys to the dict...

Related to this, I want to keep `fc_type` as a dict in MPTBlock, ffn, and attention, but I need a good way to set the default value for this dict since the `fc_type` arg will default to `None`. Importing the `fc_type_defaults` dict from `configuration_mpt.py` gives me a circular import error though. Is there a different place where we put constants that I can move the default dicts to, or should I just create a new file?

Reply:
I'll just implement what I'm talking about above and get feedback lol